craft_cli/messages.py

```diff
@@ -806,6 +807,27 @@ def confirm(self, prompt: str, *, default: bool = False) -> bool:
             return False
         return default
 
+    @_active_guard()
+    def prompt(self, prompt: str, *, hide: bool = False) -> str:
+        """Prompt user for input.
+
+        If stdin is not a tty a CraftError is raised.
+
+        :param hide: hide user input if True.
+        :returns: value that was provided by user.
+        :raises: CraftError if shell is not interactive or input is empty
```

Comment:
```suggestion
        :raises: CraftError if shell is not interactive or input is empty.
```

Reply:
Done in https://github.com/canonical/craft-cli/pull/315/commits/2a95dc594636024e7702aa70b84c8dd9877111ea